### PR TITLE
Unify algorithm unit tests to avoid code duplication

### PR DESCRIPTION
**Pull request by masterleinad**
This follows the same approach we are taking for `core/unit_test`.

### CODE REVIEW

File: algorithms/unit_tests/CMakeLists.txt [hunk 1]
```diff
@@ -33,57 +33,42 @@ IF(CMAKE_CXX_CLANG_TIDY)
...
-)
+SET(ALGORITHM UnitTestMain.cpp)
 
 IF(Kokkos_ENABLE_OPENMP)
-  LIST( APPEND SOURCES
-    TestOpenMP.cpp
+  LIST(APPEND ALGORITHM_SOURCES
     TestOpenMP_Sort1D.cpp
     TestOpenMP_Sort3D.cpp
     TestOpenMP_SortDynamicView.cpp
```
Comment (masterleinad) [inline algorithms/unit_tests/CMakeLists.txt:42]: `TestOpenMP.cpp` is merged into `TestOpenMP_Sort1D.cpp` and `TestOpenMP_Randonm.cpp` is the test corresponding to `TestBACKEND.cpp` for the other backends.

File: algorithms/unit_tests/TestCuda_Category.hpp [hunk 2]
```diff
@@ -42,36 +42,14 @@
 //@HEADER
 */
 
-#include <Kokkos_Macros.hpp>
-#ifdef KOKKOS_ENABLE_OPENMP
+#ifndef KOKKOS_TEST_CUDA_HPP
```
Comment (masterleinad) [inline algorithms/unit_tests/TestCuda_Category.hpp:45]: These files are verbatim copied from `core/unit_test`.

**Comment by masterleinad** [comment]
We could save a lot of LOC if we are to combine the `Kokkos_BACKEND_Category.hpp` for `algorithms`, `containers` and `core` somehow.

**Review by dalg24 — APPROVED** [review]
I see the scope of this PR creeped quire a bit.  I am sure we can improve even further but let's not go too crazy.

**Comment by masterleinad** [comment]
I would also be happy to split the pull request and combine all the `*Category*` files in a separate one if preferred.

**Review by DavidPoliakoff — APPROVED** [review]
Personally I wouldn't worry about splitting it up, this looks like a good bit of work

**Comment by masterleinad** [comment]
Retest this please.

**Pull request merged**